### PR TITLE
Flask app

### PR DESCRIPTION
**Pull request by davidyum**
Tried to keep the flask api example simple and in one file but let me know if we want a more "full" api as an example and add more stuff.

### CODE REVIEW

File: examples/flask_app/app.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import pcf
+
+from flask import Flask
+from pcf.core import State
+from pcf.quasiparticle.aws.ec2_route53.ec2_route53 import EC2Route53
+
+app = Flask(__name__)
+
+#We read the json configuration of our desired quasiparticle
+with open('higgs_example_config.json', 'r') as f:
```
Comment (anovis) [inline examples/flask_app/app.py]: remove higgs references

File: examples/flask_app/higgs_example_config.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+ec2_route53_example_definition = {
+    "pcf_name": "pcf-example",  
+    "flavor": "ec2_route53",  
+    "particles": [{
+        "flavor": "ec2_instance",  
+        "multiplier": 2,
```
Comment (anovis) [inline examples/flask_app/higgs_example_config.json]: can we add optional param for user to pass in that would change multiplier.

File: examples/flask_app/higgs_example_config.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            "SecurityGroupIds": [
+                "sg-11111111",
+                "sg-22222222"
+            ],
+            "SubnetId": "subnet-11111111",  
+            "UserData": "echo abc123",
+            "IamInstanceProfile": {
+                "Arn": "arn:aws:iam::111111111111:instance-profile/someRole"
+            },
+            "BlockDeviceMappings": [
```
Comment (anovis) [inline examples/flask_app/higgs_example_config.json]: dont need this for example

File: examples/flask_app/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+flask
+pcf
```
Comment (anovis) [inline examples/flask_app/requirements.txt]: lets version these

File: examples/flask_app/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# PCF-Flask-API
+The purpose of this app is to show how to use PCF in a Flask api to spin up cloud resources. In this particular example we will utilize Capital One's Particle Cloud Framework to deploy ec2-route53 quasiparticle.
```
Comment (anovis) [inline examples/flask_app/README.md]: the purpose of this example is to show one of multiple ways to use Particle Cloud Framework (PCF) to deploy cloud infrastructure which in this case is using a Flask api. In this particular example we deploy the ec2-route53 quasiparticle which creates a route53 record and adds the desired number on ec2 instances to the record set. to use in your environment  simply change some of the configurations parameters in the quasiparticle definition.

File: examples/flask_app/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Run Flask App Locally with Docker
+------------
+
+`docker build -t pcf_flask_api .`
+
+`docker run -it -p 5000:5000 pcf_flask_api `
+
+
+Run Flask App Locally with Docker
```
Comment (anovis) [inline examples/flask_app/README.md]: interact with Flask App Locally?

File: examples/flask_app/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+------------
+Once we have the flask app running locally, we can hit the endpoint to spin up the quasiparticle. 
+First, set the desired configuration in higgs_example_config.json file.
+
+To spin up the ec2-route53 quasiparticle with the desired configuration:
+
+`curl -X POST http://localhost:5000/higgs`
```
Comment (anovis) [inline examples/flask_app/README.md]: change to pcf

**Review by anovis — CHANGES_REQUESTED** [review]

File: examples/flask_app/app.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+#POST /pcf endpoint creates the EC2 Route53 Quasiparticle with the desired configuration
+@app.route('/pcf', methods=['POST'])
+def create():
+    ec2_route53_quasiparticle.set_desired_state(State.running)
+
+    if request.data:
+        payload = json.loads(request.data)
+        multiplier = payload.get('ec2_multiplier')
+        pcf_config['particles'][0]['multiplier'] = multiplier
+        ec2_route53_quasiparticle = EC2Route53(pcf_config)
```
Comment (anovis) [inline examples/flask_app/app.py:29]: this is only defined here, so other endpoints dont work. the other endpoints should take in the pcf name and unique keys so you can get state and delete.

File: examples/flask_app/app.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+with open('pcf_example_config.json', 'r') as f:
+    pcf_config = json.load(f)
+
+#Here we initialize the desired state definitions
+ec2_route53_quasiparticle.set_desired_state(State.running)
+
+#GET /pcf endpoint returns the current state of our EC2 Route53 Quasiparticle
+@app.route('/pcf', methods=['GET'])
+def get_pcf_status():
+    return str(ec2_route53_quasiparticle.get_state())
```
Comment (anovis) [inline examples/flask_app/app.py:20]: ec2_route53_quasiparticle not defined here...should create definition with only unique identifier to get state.

File: examples/flask_app/app.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        ec2_route53_quasiparticle.apply(sync=True)
+    except Exception as e:
+        raise e
+
+    return str(ec2_route53_quasiparticle.get_state())
+
+#DELETE /pcf endpoint deletes the EC2 Route53 Quasiparticle
+@app.route('/pcf', methods=['DELETE'])
+def delete():
+    ec2_route53_quasiparticle.set_desired_state(State.terminated)
```
Comment (anovis) [inline examples/flask_app/app.py:43]: ec2_route53_quasiparticle not defined here...should create definition with only unique identifier to get Quasiparticle.

**Review by anovis — CHANGES_REQUESTED** [review]

**Comment by davidyum** [comment]
@anovis let me know if you get a chance to take a look at this one. Made the fixes

**Review by anovis — APPROVED** [review]

**Pull request merged**